examples/pytorch/gat/train.py

```diff
@@ -53,37 +53,38 @@ def __init__(self,
         self.residual = residual
         if residual:
             if in_dim != out_dim:
-                self.residual_fc = nn.Linear(in_dim, num_heads * out_dim, bias=False)
-                nn.init.xavier_normal_(self.fc.weight.data, gain=1.414)
+                self.res_fc = nn.Linear(in_dim, num_heads * out_dim, bias=False)
+                nn.init.xavier_normal_(self.res_fc.weight.data, gain=1.414)
             else:
-                self.residual_fc = None
+                self.res_fc = None
 
     def forward(self, inputs):
         # prepare
-        h = inputs
+        h = inputs  # NxD
         if self.feat_drop:
             h = self.feat_drop(h)
-        ft = self.fc(h).reshape((h.shape[0], self.num_heads, -1))
-        head_ft = ft.transpose(0, 1)
-        a1 = torch.bmm(head_ft, self.attn_l).transpose(0, 1)
-        a2 = torch.bmm(head_ft, self.attn_r).transpose(0, 1)
+        ft = self.fc(h).reshape((h.shape[0], self.num_heads, -1))  # NxHxD'
+        head_ft = ft.transpose(0, 1)  # HxNxD'
+        a1 = torch.bmm(head_ft, self.attn_l).transpose(0, 1)  # NxHx1
+        a2 = torch.bmm(head_ft, self.attn_r).transpose(0, 1)  # NxHx1
         if self.feat_drop:
             ft = self.feat_drop(ft)
         self.g.ndata.update({'ft' : ft, 'a1' : a1, 'a2' : a2})
         # 1. compute edge attention
         self.g.apply_edges(self.edge_attention)
-        # 2. compute two results, one is the node features scaled by the dropped,
-        # unnormalized attention values. Another is the normalizer of the attention values.
+        # 2. compute two results: one is the node features scaled by the dropped,
+        # unnormalized attention values; another is the normalizer of the attention values.
         self.g.update_all([fn.src_mul_edge('ft', 'a_drop', 'ft'), fn.copy_edge('a', 'a')],
                           [fn.sum('ft', 'ft'), fn.sum('a', 'z')])
         # 3. apply normalizer
-        ret = self.g.ndata['ft'] / self.g.ndata['z']
+        ret = self.g.ndata['ft'] / self.g.ndata['z']  # NxHxD'
         # 4. residual
         if self.residual:
-            if self.residual_fc:
-                ret = self.residual_fc(h) + ret
+            if self.res_fc is not None:
+                resval = self.res_fc(h).reshape((h.shape[0], self.num_heads, -1))  # NxHxD'
```

Comment:
If my previous comment on `self.res_fc` is correct, this should probably be `self.res_fc(h).unsqueeze(1)`.